src/umpire/resource/HipConstantMemoryResource.hpp

```diff
@@ -0,0 +1,58 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_HipConstantMemoryResource_HPP
+#define UMPIRE_HipConstantMemoryResource_HPP
+
+#include "umpire/resource/MemoryResource.hpp"
+
+#include "umpire/util/AllocationRecord.hpp"
+#include "umpire/util/Platform.hpp"
+
+#include <hip/hip_runtime.h>
+
+__constant__ char umpire_internal_device_constant_memory[64*1024];
```

Comment:
Is this configurable?  Or is this fixed to 64*1024?

Reply:
I hadn't thought to make this configurable. Let me check and see whether the constant memory amounts change significantly on different AMD GPUs.

Reply:
This is probably fine for now.  I was just checking.

Reply:
So here's a more lengthy explanation of the "constant" memory cache in AMD hardware: Despite the legacy name, this is actually a read+write cache. It uses scalar memory instructions to efficiently 'broadcast' a memory load to all the threads in a compute unit (CU). On the CUs, there is a dedicated cache for this memory and if a memory request misses in this cache it will spill to L2 and then to global memory. So you can allocate GBs of "constant" memory if you'd like. However, requests may then spill. I think for Vega architectures this hardware cache is 16KB.